File: SDKs/jellyfin.local/v1/go/pkg/models/operations/getsimilaritems.go
package operations

import (
	"openapi/pkg/models/shared"
)

type GetSimilarItemsPathParams struct {
	ItemID string `pathParam:"style=simple,explode=false,name=itemId"`
}

type GetSimilarItemsQueryParams struct {
	ExcludeArtistIds []string                `queryParam:"style=form,explode=true,name=excludeArtistIds"`
	Fields           []shared.ItemFieldsEnum `queryParam:"style=form,explode=true,name=fields"`
	Limit            *int32                  `queryParam:"style=form,explode=true,name=limit"`
	UserID           *string                 `queryParam:"style=form,explode=true,name=userId"`
}

type GetSimilarItemsSecurity struct {
	CustomAuthentication shared.SchemeCustomAuthentication `security:"scheme,type=apiKey,subtype=header"`
}

type GetSimilarItemsRequest struct {
	PathParams  GetSimilarItemsPathParams
	QueryParams GetSimilarItemsQueryParams
	Security    GetSimilarItemsSecurity
}

type GetSimilarItemsResponse struct {
	BaseItemDtoQueryResult *shared.BaseItemDtoQueryResult
	ContentType            string
	StatusCode             int64
}

File: SDKs/jellyfin.local/v1/typescript/src/sdk/models/operations/getsimilaritems.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class GetSimilarItemsPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=itemId" })
  itemId: string;
}


export class GetSimilarItemsQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=excludeArtistIds" })
  excludeArtistIds?: string[];

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=fields" })
  fields?: shared.ItemFieldsEnum[];

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=limit" })
  limit?: number;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=userId" })
  userId?: string;
}


export class GetSimilarItemsSecurity extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "security, scheme=true;type=apiKey;subtype=header" })
  customAuthentication: shared.SchemeCustomAuthentication;
}


export class GetSimilarItemsRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: GetSimilarItemsPathParams;

  @SpeakeasyMetadata()
  queryParams: GetSimilarItemsQueryParams;

  @SpeakeasyMetadata()
  security: GetSimilarItemsSecurity;
}


export class GetSimilarItemsResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  baseItemDtoQueryResult?: shared.BaseItemDtoQueryResult;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;
}
